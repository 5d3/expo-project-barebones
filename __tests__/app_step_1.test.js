import React from 'react';
import { View, Text } from 'react-native';
import { shallow, render, mount } from 'enzyme';

import App from '../app/app_step_1';

afterAll(() => {
  console.log('All tests finished');
});

afterEach(() => { /* do something */ });
beforeAll(() => { /* do something */ });
beforeEach(() => { /* do something */ });

describe("<App />", function() {
  const wrapper = shallow(<App />);

  it('should render an App component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('  should have a View component wrapper', () => {
    expect(wrapper.type()).toEqual(View);
  });

  it('  should have flex equal to 1', () => {
    expect(wrapper.props().style.flex).toEqual(1);
  });

  it('  should have only one child', () => {
    expect(wrapper.children()).toHaveLength(1);
  });

  it('    should be a View component', () => {
    // expect(wrapper.children().first().type()).toBe(View);
    expect(wrapper.childAt(0).type()).toBe(View);
  });

  it('    should have two children', () => {
    expect(wrapper.childAt(0).children()).toHaveLength(2);
  });

  it('      should be two Text components', () => {
    wrapper.childAt(0).children().forEach(child => {
        expect(child.type()).toBe(Text)
    });
  });

  it('        should have its first Text component with "Welcome to rmotr\'s workshop!" message', () => {
    expect(wrapper.childAt(0).childAt(0).props().children).toBe('Welcome to rmotr\'s Workshop!');
  });

  it('        should have its first Text component with fontSize 20', () => {
    expect(wrapper.find(Text).get(0).props.style.fontSize).toBe(20);
  });

  it('        should have its first Text component with color \'#2c3e50\'', () => {
    expect(wrapper.find(Text).get(0).props.style.color).toBe('#2c3e50');
  });

  it('        should have its first Text component with fontWeight \'bold\'', () => {
    expect(wrapper.find(Text).get(0).props.style.fontWeight).toBe('bold');
  });

  it('        should have its second Text component with "(Open up main.js to start working)" message', () => {
    expect(wrapper.find(Text).get(1).props.children).toBe('(Open up main.js to start working)');
  });

  it('        should have its second Text component with color #34495e', () => {
    expect(wrapper.find(Text).get(1).props.style.color).toBe('#34495e');
  });
});
