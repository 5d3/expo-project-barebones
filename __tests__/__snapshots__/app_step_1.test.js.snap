// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render an App component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "flex": 1,
          }
    }
>
    <View
        style={
            Array [
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#eee",
                  "justifyContent": "center",
                  "padding": 10,
                },
                Object {
                  "flex": 1,
                },
              ]
        }
    >
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#2c3e50",
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
            }
        >
            Welcome to rmotr's Workshop!
        </Text>
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#34495e",
                    "fontStyle": "italic",
                    "paddingBottom": 5,
                  }
            }
        >
            (Open up main.js to start working)
        </Text>
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "flex": 1,
                }
      }
>
      <View
            style={
                  Array [
                        Object {
                          "alignItems": "center",
                          "borderBottomWidth": 1,
                          "borderColor": "#eee",
                          "justifyContent": "center",
                          "padding": 10,
                        },
                        Object {
                          "flex": 1,
                        },
                      ]
            }
      >
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#2c3e50",
                              "fontSize": 20,
                              "fontWeight": "bold",
                            }
                  }
            >
                  Welcome to rmotr's Workshop!
            </Text>
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#34495e",
                              "fontStyle": "italic",
                              "paddingBottom": 5,
                            }
                  }
            >
                  (Open up main.js to start working)
            </Text>
      </View>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
>
          <View
                    style={
                              Array [
                                        Object {
                                          "alignItems": "center",
                                          "borderBottomWidth": 1,
                                          "borderColor": "#eee",
                                          "justifyContent": "center",
                                          "padding": 10,
                                        },
                                        Object {
                                          "flex": 1,
                                        },
                                      ]
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#2c3e50",
                                                  "fontSize": 20,
                                                  "fontWeight": "bold",
                                                }
                              }
                    >
                              Welcome to rmotr's Workshop!
                    </Text>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#34495e",
                                                  "fontStyle": "italic",
                                                  "paddingBottom": 5,
                                                }
                              }
                    >
                              (Open up main.js to start working)
                    </Text>
          </View>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
>
          <View
                    style={
                              Array [
                                        Object {
                                          "alignItems": "center",
                                          "borderBottomWidth": 1,
                                          "borderColor": "#eee",
                                          "justifyContent": "center",
                                          "padding": 10,
                                        },
                                        Object {
                                          "flex": 1,
                                        },
                                      ]
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#2c3e50",
                                                  "fontSize": 20,
                                                  "fontWeight": "bold",
                                                }
                              }
                    >
                              Welcome to rmotr's Workshop!
                    </Text>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#34495e",
                                                  "fontStyle": "italic",
                                                  "paddingBottom": 5,
                                                }
                              }
                    >
                              (Open up main.js to start working)
                    </Text>
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
